CONTRIBUTING.md

```diff
@@ -0,0 +1,232 @@
+- **Explain why this enhancement would be useful** to most Movement users. You may also want to point out the other projects that solved it better and which could serve as inspiration.
+
+<!-- You might want to create an issue template for enhancement suggestions that can be used as a guide and that defines the structure of the information to be included. If you do so, reference it here in the description. -->
+
+### Your First Code Contribution
+<!-- TODO
+include Setup of env, IDE and typical getting started instructions?
+
+-->
+If you’re new to contributing to open source or unsure where to start, we’ve got you covered. We recommend the following steps for your first code contribution:
+
+- **Find a “good first issue”** – These issues are marked to help new contributors start with smaller, manageable contributions.
+- **Set up the project** – Follow the instructions in the repository’s [README](https://github.com/movementlabsxyz/movement/blob/main/README.md) to set up the project locally.
+- **Get in touch** – Comment on the issue you’re interested in working on. Our maintainers will assist if you need any guidance.
+- **Make the changes** – Start small; fix a bug, update documentation, or work on a minor feature.
+- **Submit a Pull Request (PR)** – Once you’re ready, open a PR with a clear description of your changes.
+
+We’ll review your code, offer feedback, and help with any questions.
+
+### Improving The Documentation
+<!-- TODO
+Updating, improving and correcting the documentation
+
+-->
+Documentation improvements are always appreciated and highly valuable to the project. Here are some ways to improve documentation:
+
+- **Correcting typos or grammar** – Small changes can make a big difference in readability.
+- **Clarifying confusing sections** – If anything seems unclear to you, it likely will to others as well.
+- **Adding examples** – Code examples are incredibly helpful in showing how things work in practice.
+- **Updating outdated information** – Ensure everything reflects the current state of the project.
+
+When making documentation updates, remember to follow the **Styleguides** below.
+
+### Styleguides
+#### Code Style
+
+For Rust, we follow the standard Rust style guidelines, which include the following:
+
+- **Indentation**: Use 4 spaces for indentation, no tabs.
+- **Naming Conventions**: 
+  - Use `snake_case` for variables, functions, and file names.
+  - Use `PascalCase` for structs, enums, and traits.
+  - Use `ALL_CAPS` with underscores for constants.
+- **Module Organization**: Keep modules organized logically, typically grouping related structs, enums, and functions together.
+- **Error Handling**: Prefer `Result` and `Option` types over panics where possible.
+- **Documentation Comments**: Use `///` for public-facing functions and modules, and `//` for inline comments.
```

Comment:
this could do with an example to e.g. standardise the function comments

Reply:
These are already standardized in Rust though.

Reply:
Fair point, makes sense to link to https://doc.rust-lang.org/style-guide/index.html